### PR TITLE
Ensure error is raised for playing running Cylc 7 workflow

### PR DESCRIPTION
**Pull request by datamel**
Closes https://github.com/cylc/cylc-flow/issues/5317

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Tests are included (or explain why tests are not needed).
- [ ] `CHANGES.md` entry included if this is a change that can affect users
- [ ] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [ ] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
Could you rebase onto 8.1.x please.

**Comment by wxtim** [comment]
I don't think that you've rebased onto 8.1.x correctly - you will need to do `cylc rebase upstrea/8.1.x -i` and remove the commits unrelated to the PR.

**Comment by datamel** [comment]
> * I don't think that you've rebased onto 8.1.x correctly - you will need to do `cylc rebase upstrea/8.1.x -i` and remove the commits unrelated to the PR.

I think once the merge to 8.1.x has been sorted the commits should go

> Can you do a changelog entry?

Yes but I'll wait until someone can reproduce. No point in fixing something that may not be broken.
To reproduce I set off a long-ish running workflow with Cylc 7, then activated Cylc 8 env (I pip installed too), the played the workflow that was running with cylc 7 but this time with Cylc 8. I got KeyError traceback, this just raises a slightly neater error.

**Comment by oliver-sanders** [comment]
When you rebase back from master to an earlier branch (e.g. 8.1.x) you may drag some unwanted commits with you. There are four commits on this branch:

https://github.com/cylc/cylc-flow/pull/5323/commits

The unwanted commits can be safely removed with `git rebase -i`.

File: cylc/flow/workflow_files.py [hunk 1]
```diff
@@ -44,6 +44,7 @@
 from cylc.flow.cfgspec.glbl_cfg import glbl_cfg
 from cylc.flow.exceptions import (
     CylcError,
+    CylcVersionError,
```
Comment (datamel) [inline cylc/flow/workflow_files.py:47]:
```suggestion
```

**Comment by oliver-sanders** [comment]
I think `detect_old_contact` file fails for real Cylc 7 contact files.

https://github.com/cylc/cylc-flow/pull/5313#discussion_r1084285970

The solution on #5313 was to catch the error, log a warning about Cylc 8 not being able to start Cylc 7 workflows, then raise the error.

**Comment by datamel** [comment]
> I think `detect_old_contact` file fails for real Cylc 7 contact files.

I am getting traceback after running with cylc 7:
```
Traceback (most recent call last):
  File "~/cylc-flow/cylc/flow/workflow_files.py", line 507, in detect_old_contact_file
    old_pid: str = contact_data[ContactFileFields.PID]
KeyError: 'CYLC_WORKFLOW_PID'
 ```

My cylc 7 contact file has a field `CYLC_SUITE_PROCESS`, not the  'CYLC_WORKFLOW_PID' this method is looking for, hence the key error.

> 
> [#5313 (comment)](https://github.com/cylc/cylc-flow/pull/5313#discussion_r1084285970)
> 
> The solution on #5313 was to catch the error, log a warning about Cylc 8 not being able to start Cylc 7 workflows, then raise the error.

Happy to close this and have it fixed on #5313, I  picked up the bug while reviewing it so makes sense to put the fix in there. 
For me, this traceback is still occurring on #5313 in this scenario. 

@wxtim was unable to reproduce this bug so can you confirm this is happening for you @oliver-sanders?

**Comment by datamel** [comment]
I think the error is only being raised on `vr`, not just play on https://github.com/cylc/cylc-flow/pull/5313. Whereas this bug I am referring to is when you just `play` a running cylc 7 workflow.

**Comment by oliver-sanders** [comment]
> Still happy for this to be fixed on @wxtim's branch.

Tim's branch fixes it for `cylc vr` but I'm not sure if it fixes it for `cylc play`.

**Comment by oliver-sanders** [comment]
> Here is the traceback from the vr command on https://github.com/cylc/cylc-flow/pull/5313 with a running cylc7 workflow.

IMO this traceback is a good enough solution. So long as the "Note Cylc 8 can't restart Cylc 7 workflows" notice is in there in bold red. This traceback could result from other things and it's probably more bother than it's worth trying to tell the difference between a Cylc 7/8 contact file and all that junk.

So probably just need to make sure other commands i.e. `play` (and possibly others?) aren't affected. (note `cylc scan` has its own logic for filtering out Cylc 7 workflows).

**Comment by datamel** [comment]
Closing this after the above discussion.

**Pull request closed without merging**